### PR TITLE
Support filtering by session ID #126

### PR DESCRIPTION
**Pull request by MrWilhelm**
Ability to query archives by session id. #126 

`OpenTokAdapter.client.archives.all(session_id: 'SESSIONID')`

### CODE REVIEW

**Comment by MrWilhelm** [comment]
@aoberoi @jeffswartz Can you review this?

**Comment by kouno** [comment]
I opened https://github.com/opentok/OpenTok-Ruby-SDK/pull/132 not looking at other PRs so I don't mind if @MrWilhelm gets feedback on this one and if we close mine, but could we get something done around this issue?

@aiham @thepatrick might be able to help since they recently worked on the project 😄 ? Thanks everyone!

**Comment by aiham** [comment]
@kouno @MrWilhelm Sure. I'll take a look at this.

File: spec/cassettes/OpenTok_Archives/when_many_archives_are_created/should_return_session_archives.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+http_interactions:
+- request:
+    method: get
+    uri: https://api.opentok.com/v2/project/123456/archive?sessionId=SESSIONID
+    body:
+      encoding: US-ASCII
+      string: ''
+    headers:
+      X-Tb-Partner-Auth:
```
Comment (aiham) [inline spec/cassettes/OpenTok_Archives/when_many_archives_are_created/should_return_session_archives.yml]: Can you replace `X-Tb-Partner-Auth` with `X-Opentok-Auth` like the other cassettes. See: https://github.com/opentok/OpenTok-Ruby-SDK/blob/master/spec/cassettes/OpenTok_Archives/should_create_archives.yml#L12

File: lib/opentok/archives.rb [hunk 2]
```diff
@@ -100,7 +100,9 @@ def find(archive_id)
     # @return [ArchiveList] An ArchiveList object, which is an array of Archive objects.
```
Comment (aiham) [inline lib/opentok/archives.rb]: Can you add a `@param` comment above for the `:session_id` option?

**Comment by aiham** [comment]
No response to change request. Implemented in #143.

**Pull request closed without merging**